code/Model/RedirectorPageController.php

```diff
@@ -11,19 +12,28 @@ class RedirectorPageController extends PageController
 {
     private static $allowed_actions = ['index'];
 
+    /**
+     * @var bool Should respond with HTTP 404
+     */
+    private static $should_respond_404 = false;
+
     /**
      * Check we don't already have a redirect code set
      *
      * @param  HTTPRequest $request
      * @return \SilverStripe\Control\HTTPResponse
+     * @throws HTTPResponse_Exception
      */
     public function index(HTTPRequest $request)
     {
         /** @var RedirectorPage $page */
         $page = $this->data();
         if (!$this->getResponse()->isFinished() && $link = $page->redirectionLink()) {
-            $this->redirect($link, 301);
+            return $this->redirect($link, 301);
+        } elseif ($page->LinkToID && $this->config()->get('should_respond_404') === true) {
```

Comment:
This condition is a bit cryptic for me @ssmarco, can we refactor it so it's cleaner what/why it's testing? I would expect to test something like `!$page->LinkTo()->exists()` here.

Reply:
Or can we add a comment why we're not doing that and whether it needs to check for something else or relying on the previous condition etc.